webpack/components/EditableTextInput/EditableTextInput.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, Fragment } from 'react';
+import { TextInput, TextArea, Text, TextVariants, Button, Split, SplitItem } from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, EditIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import Loading from '../Loading';
+import './editableTextInput.scss';
+
+const EditableTextInput = ({
+  onEdit, value, textArea, editable, label,
+}) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [working, setWorking] = useState(false);
+  const onSubmit = () => {
+    setWorking(true);
+    onEdit(inputValue);
+    setWorking(false);
+    setEditing(false);
+  };
+  const onClear = () => {
+    setInputValue(value);
+    setEditing(false);
+  };
+
+  const textInput = () => {
+    const sharedProps = {
+      value: inputValue,
+      onChange: v => setInputValue(v),
+    };
+
+    if (textArea) {
+      return <TextArea {...sharedProps} aria-label={`text area ${label}`} />;
+    }
+    return <TextInput {...sharedProps} type="text" aria-label={`text input ${label}`} />;
+  };
+
+  if (working) return <Loading size="sm" />;
+  return (
+    <Fragment>
+      {editing ?
```

Comment:
Prefer conditional rendering to avoid the long ternary:
```javascript
<Fragment>
  {editing &&
    <Split>
     ...
   </Split>
  }
  {!editing &&
  <Split>
     ...
   </Split>
  }
</Fragment>
```

Reply:
I'm not sure if this is the best practice, do you see any examples or docs with this way of writing? I would rather change to an if..else than value...!value.

I think using ternaries is more just a matter of style preference. React docs provide both ternary and if..else as an example  so both seem to be acknowledged as legitimate values: https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator.

If you feel strongly about it, I can change to if..else, but I would like to stick with the ways the docs provide

Reply:
if..else would have to be outside the render function, as shown in the top of the page

Reply:
React doc recommending this pattern [here](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

The React docs also illustrate using a long [ternary](https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator), but caution that it's "less obvious what's going on"

Personally I strongly prefer the former (but you knew that ;)

Reply:
> React doc recommending this pattern [here](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

That is for an inline if logic, not if...else logic like we have here.

I've never seen the `&&` used for if...else logic and I don't see the React docs recommending that. I wouldn't do `value...!value` in normal javascript, so I can't really see the argument for it. I'm going to keep it as is, but I'm open to other opinions or examples.

Reply:
My understanding of the choices the docs give for if..else are:
- ternary operator
- if..else statement outside of render/return statement
- break into separate components

Reply:
I would go for breaking into separate components before I had a ternary that long.

Reply:
let me know what you think of what I've updated to

Reply:
Totally fine with `if` with early returns.  :+1: